### DISCUSSION TITLE
Configuring Quartz as BackgroundService - specifically Plugins

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#1607] [Q&A]
Hi, I've just started working with Quartz.NET, so I'm a novice with it. I've created a .NET 6.0 BackgroundService app and am successfully running the code from Quartz.Examples.Worker.

I want to be able to configure things like the LoggingTriggerHistoryPlugin (and job history). I've added the following to my appsettings.json:

`{
  "Quartz": {
    "quartz.scheduler.instanceName": "Quartz Worker Service Sample Scheduler From appsettings",
    "quartz.scheduler.instanceId": "Scheduler ID From appsettings",
    "quartz.plugin.triggHistory.type": "Quartz.Plugin.History.LoggingTriggerHistoryPlugin, Quartz.Plugins"
  }
}`

But when I run the app it doesn't seem like any of this configuration information is being used. I say that because when I inspect q.Properties from within services.AddQuartz, there are 16 keys there but none of them are the ones I entered above.

Also, I have nlog configured and I see INFO logging messages from Quartz and messages logged from ExampleJob.cs, but I don't see anything that looks like the default messaged from either LoggingTriggerHistoryPlugin  or LoggingJobHistoryPlugin 

So my questions are:
Q1. In order to configure this when running as a service, do I need to implement it like in this link, so I can set the properties namevaluecollection on the StdSchedulerFactory?  https://andrewlock.net/creating-a-quartz-net-hosted-service-with-asp-net-core/  This person isn't setting config settings in the article, I'm just wondering if I need to create my own scheduler rather than use services.AddQuartz.

Q2. Do I have to do anything special to log Trigger and Job history? I have the Quarts.Plugin nuget package installed. NLog is logging other things. If I configure those plugs ins (which I don't know where to do), should they log to my NLog targets?

Thanks Much.
Brent

**Comment by lahma** [comment]
Have you checked the ASP.NET Core sample? It's more complete and contains configuration to read from multiple sources. It also has Serilog configured, NLog probably would follow same patterns.

**Reply by ghost** [reply]
I have NLog working fine... My question was about configuring Quartz and viewing things logged from: Quartz.Plugin.History.LoggingTriggerHistoryPlugin and Quartz.Plugin.History.LoggingJobHistoryPlugin

I'll go to stackoverflow, thanks.

**Reply by lahma** [reply]
I think your question is valid here too, but again, the aspnet core sample shows configuration properties from multiple sources and you can always access the raw key value pairs inside AddQuartz too if configuration via code is option.

**Reply by ghost** [reply]
Ah, I hadn't seen .SetProperty. I've set the properties for the plugins, in code, and now they are logging successfully. Very sorry for my novice level question. The quartz project is very impressive, just a lot to learn when a person is rushing. THANK YOU!
